src/settled-cash/CrossMarginCashEngine.sol

```diff
@@ -282,6 +293,62 @@ contract CrossMarginCashEngine is
         return _getMinCollateral(account);
     }
 
+    /**
+     * @notice  grant or revoke an account access to all your sub-accounts based on a signed message
+     * @dev     expected to have a valid signature signed with account private key
+     * @param   _account account which grants the access
+     * @param   _actor account that is authorized to access the margin account
+     * @param   _allowedExecutions how many times the account is authorized to update your accounts.
+     *          set to max(uint256) to allow permanent access
+     * @param   _v signature v
+     * @param   _r signature r
+     * @param   _s signature s
+     */
+    function setAccountAccess(address _account, address _actor, uint256 _allowedExecutions, uint8 _v, bytes32 _r, bytes32 _s)
```

Comment:
This function signature is very confusing when you consider `setAccountAccess` in the base engine.

You are changing `account` to be a completely different entity in this function. One whose function is very unclear to me. Why is this needed?

Why is `engine` included? I suspect its to not use this same message against other engines? If that's the case then `verifyingContract` already takes care of this for us.

Lets change the function name and signing object to `PermitAccountAccess`